jekyll/_cci2/optimization-cookbook.md

```diff
@@ -0,0 +1,153 @@
+
+### Optimization Recipes
+
+The table below lists the different optimization tasks that can be performed using the CircleCI platform.
+
+Recipe | Description 
+------------|-----------
+Single-Threading Builds (Queueing) | This section describes how you can use a specific single-threading (queueing) orb to ensure proper job and build completion without concurrency (queueing subsequent builds or jobs).
+Caching |  This section describes how you can use different caching strategies in your implemenations to optimize your builds and workflows.
+Workflow Optimizations | This section describes how you can optimize your workflows.
+
+## Single-Threading Builds (Queuing)
+
+One of the most common tasks you may encounter when using the CircleCI platform for builds and jobs is managing multiple jobs and builds simultaneously to ensure your operations do not fail because of system timeouts. This becomes especially important when you have multiple contributors and committers working in the same environment. Because the CircleCI platform was designed to handle multiple tasks simultaneously without performance degradation or latency, concurrency may sometimes become an issue if there are a large number of jobs being queued, waiting for a previous job to be completed before the new job can be initiated, and the system timeout is set too low. In this case, one job will be completed, and other jobs may fail due to this timeout setting.
+
+To better optimize builds and jobs and prevent concurrency and subsequent jobs failing because of timeout, CircleCI has developed a single-threading (queueing) orb that specifically addresses these performance issues. By invoking this orb, you can greatly improve overall job and build performance and prevent concurrency.
+
+**Note**For more detailed information about the CircleCI Queueing orb, refer to the following CircleCI pages:
+- Queueing and Single Threading Overview - https://github.com/eddiewebb/circleci-queue
+- CircleCI Queueing Orb - https://circleci.com/orbs/registry/orb/eddiewebb/queue#quick-start
+
+### Setting Up and Configuring Your Environment to use the CircleCI Platform and CircleCI Orbs
+
+To configure your environment for the CircleCI platform and CircleCI orbs, follow the steps listed below.
+
+1) Use CircleCI `version 2.1` at the top of your `.circleci/config.yml` file.
+
+`version: 2.1`
+
+2) If you do not already have Pipelines enabled, you'll need to go to **Project Settings -> Advanced Settings** to enable pipelines.
+
+3) Add the `orbs` stanza below your version, invoking the orb:
+
+`orbs:
+  queue: eddiewebb/queue@1.1.2`
+
+4) Use `queue` elements in your existing workflows and jobs.
+
+5) Opt-in to use of third-party orbs on your organization’s **Security Settings** page.
```

Comment:
If we're recommending it in docs, we should probably make it an orb that doesn't need to be opt-ed in?

Reply:
I agree, though I personally don't know anything about this orb and its use case.

Reply:
fixed

Reply:
Does this opt in step need to come before step 4?